### DISCUSSION TITLE
Use Transcode to Downscale Video

### DISCUSSION THREAD

**Discussion by tibmeister** [koush/scrypted#1387] [Q&A] [answered]
Got a couple of cameras that are remote, and they insist on only having higher resolutions in the main stream, and only 640x480 on the sub-stream.  At that location, I have Scrypted installed pulling the camera streams there locally.  From there, I "pull" those streams from my main location, using another instance of Scrypted, and this occurs over a IPSeC tunnel between the two locations.  This setup is so that the cameras only have one system pulling the stream, and I can consume the stream on multiple other systems without killing the cameras.  At the main location I have a couple of systems, including HKSV, pulling the streams from those remote cameras.  I also have a local instance of Home Assistant running at the remote location with the cameras pulling the video stream from Scrypted into a dashboard.

So, it's like this:

Camers > Scrypted (Reolink Plugin) >> IPSec Tunnel >> Scrypted (RTSP Camera) > Frigate/HKSV

Now that the architecture is out of the way...

On these cameras, since I can't reduce the resolution below 2304x1296, and the camera seems to always revert back to 2560x1920, I would like to scale the video down to 1920x1440 or 1280x960, keeping the 4:3 AR while using less bandwidth over the IPSeC tunnel.

I enabled the Transcoding plugin for a camera, set the stream (Local Stream) to transcode, and in the "Video Filter Arguments" I have entered '-vf scale=1920:1440', and this doesn't seem to work because the stream at the other end is still seeing the original 2560x1920 resolution at 2048k bit rate.  This tells me the Transcoding plugin either isn't working, or there's something I need to do differently on the remote end when referencing the stream from Scrypted, which for Local Stream would be the main stream, which the Rebroadcast URL is rtsp://localhost:32993/072fcd87b6c04305.

The documentation on the Transcoding plugin is very sparse, so my assumption is that the Rebroadcast URL for the Local Stream should be the transcoded video stream since the plugin is enabled.  If this is not correct, please provide more information on how the Transcoding plugin should work and what I need to adjust to get the lower stream.

**Comment by koush** [comment, marked as answer]
There's currently no easy way to do this. You can use the rebroadcast URL on the camera to create a new camera device, and then transcode that on input using ffmpeg parser.

**Reply by tibmeister** [reply]
So what exactly does the Transcode plugin do then?  Also, if I'm doing it on the input, how would one set that up?  Guessing since I'm using the Reolink plugin it's not possible but if I switch to using ONVIF or RTSP, I may have options?

**Reply by koush** [reply]
It's used to transcode video on output to other destinations in scrypted. I'm not sure thats the correct approach anymore, and will very likely remove it soon.
